pkg/transform/internal/cache/cache.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"sync"
+
+	"github.com/bufbuild/buf/private/pkg/protoencoding"
```

Comment:
Oy, this will be a bit of a bear to untangle out of "private".  I think that's another bit of evidence that we should probably incubate this stuff inside of "private" and then figure out a path to open-sourcing it for external use as a library later.